feature-account-api/src/main/java/io/novafoundation/nova/feature_account_api/domain/model/MetaAccount.kt

```diff
@@ -252,11 +251,27 @@ fun LightMetaAccount.Type.requestedAccountPaysFees(): Boolean {
     }
 }
 
-val LightMetaAccount.Type.isProxied: Boolean
-    get() = this == LightMetaAccount.Type.PROXIED
+@OptIn(ExperimentalContracts::class)
+fun LightMetaAccount.isProxied(): Boolean {
+    contract {
+        returns(true) implies (this@isProxied is ProxiedMetaAccount)
+    }
+
+    return type == LightMetaAccount.Type.PROXIED
```

Comment:
It is better to ensure condition in contract matches the condition in the function itself. Otherwise we might face a crash if type is Proxied but account is not ProxiedMetaAccount (this could be due to some error in the account mapping logic for example)